### ISSUE TITLE
sizePerPageDropDown not working in IE 11

### ISSUE THREAD

**Issue by nperriwp** [AllenFang/react-bootstrap-table#1567]
I am using react-bootstrap-table 4.0.1, IE 11.0.9600.18762, bootstrap 3.3.7.

When I use the sizePerPageDropDown option and create my own sizePerPageDropDown control, it works fine in Chrome, but for IE 11 it seems to ignore this option completely. My sizePerPageDropDown render function creates a drop down box with the values 10, 20, 30, 40, 50. I see this fine in Chrome. However, when rendering in IE it shows the default 10, 25, 30, 50 and doesn't render anything in my function for sizePerPageDropDown. My code is below:

  renderSizePerPageDropDown(props) {
    const sizes = [10, 20, 30, 40, 50]
    const sizePerPageOptions = sizes.map(sizeItem => <option key={sizeItem} value={sizeItem}>{sizeItem}</option>)
    return (
      <div>
        <FormGroup style={{ marginBottom: "0" }}>
          <FormControl.Static style={{ display: "inline-block" }} >Size Per Page:&nbsp;&nbsp;</FormControl.Static>
          <FormControl
            componentClass="select"
            value={20}
            onChange={(event) => this.props.dispatch(doSomething(parseInt(event.target.value, 10)))}
            style={{ display: "inline-block", width: "5em" }}
          >
            {sizePerPageOptions}
          </FormControl>
        </FormGroup>
      </div>
    )
  }

  render() {
        ....

        const options = {
          sizePerPage: 20,
          sizePerPageDropDown: this.renderSizePerPageDropDown,
        }
        const TableHeaderColumns =[an array of TableHeaderColumn]

        return (
              <BootstrapTable
                ref={tableRef => { this.tableRef = tableRef }}
                data={data}
                options={options}
                tableStyle={{ border: "none" }}
                condensed
                striped
                pagination
              >
                {TableHeaderColumns}
              </BootstrapTable>
       )
}

Any idea why this doesn't work in IE 11?

**Comment by prajapati-parth** [comment]
@nperriwp Can you please check if the console shows any error? Thanks.

**Comment by AllenFang** [comment]
@nperriwp please refine your source code, it's to hard to understand.

can you try to remove `FormControl`, just use `div`.

**Comment by nperriwp** [comment]
There are no console errors.

I tried removing FormControl and just put a div in, but that did not work either on Internet Explorer.

**Comment by nperriwp** [comment]
Here is my code again. Is this better?

<pre>
renderSizePerPageDropDown(props) {
    const sizes = [10, 20, 30, 40, 50]
    const sizePerPageOptions = sizes.map(sizeItem => {sizeItem})
    return (
      &lt;div&gt;
        &lt;FormGroup style={{ marginBottom: "0" }}&gt;
          &lt;FormControl.Static style={{ display: "inline-block" }} >Size Per Page:  &lt;/FormControl.Static&gt;
          &lt;FormControl componentClass="select" value={20} style={{ display: "inline-block", width: "5em" }}
                       onChange={(event) => this.props.dispatch(doSomething(parseInt(event.target.value, 10)))} &gt;
           {sizePerPageOptions}
          &lt;/FormControl&gt;
        &lt;/FormGroup&gt;
      &lt;/div&gt;
   )
}

render() {

    const options = {
      sizePerPage: 20,
      sizePerPageDropDown: this.renderSizePerPageDropDown,
    }
    const TableHeaderColumns =[an array of TableHeaderColumn]

    return (
          <BootstrapTable
            ref={tableRef => { this.tableRef = tableRef }}
            data={data}
            options={options}
            tableStyle={{ border: "none" }}
            condensed
            striped
            pagination
          >
            {TableHeaderColumns}
          </BootstrapTable>
   )
}
</pre>

**Comment by prajapati-parth** [comment]
@nperriwp Thanks for the simplified code. I'm not so sure about this but this can be just another issue similar to #1565, if you are using react-bootstrap. Please try without using them.

**Comment by nperriwp** [comment]
I cannot try removing react-bootstrap from my project because too much of my code depends on that library. I am not using MenuItem at all. In fact, if I just use a &lt;div&gt; tag in renderSizePerPageDropDown rather than the react-bootstrap tags, I still have a problem with Internet Explorer. Also, this feature works fine in Chrome, it is just not working in Internet Explorer. #1565 is not an issue with a specific browser like this issue.

**Comment by AllenFang** [comment]
@nperriwp could you help me to try [this](https://allenfang.github.io/react-bootstrap-table/example.html#pagination) page on your IE?

**Comment by nperriwp** [comment]
@AllenFang That page works for me in IE, but none of the examples on the page use the sizePerPageDropDown option.

**Comment by AllenFang** [comment]
@nperriwp how about this page: https://allenfang.github.io/react-bootstrap-table/example.html#custom
I think the bootstrap dropdown haven't bug on IE, btw, what's the version of your bootstrap

**Comment by nperriwp** [comment]
@AllenFang I can see "Default Custom SizePerPage DropDown" working in IE, but "Fully Custom SizePerPage DropDown" does not work in IE. In Chrome for "Fully Custom SizePerPage DropDown" I see 3 blue buttons labeled 10, 25 and 30. However, in IE "Fully Custom SizePerPage DropDown" shows the default drop down of 10, 25, 30, 50.

My bootstrap is 3.3.7.

**Comment by AllenFang** [comment]
> In Chrome for "Fully Custom SizePerPage DropDown" I see 3 blue buttons labeled 10, 25 and 30

yes, that's a demo that i custom the sizePerPage component as a buttongroups

> However, in IE "Fully Custom SizePerPage DropDown" shows the default drop down of 10, 25, 30, 50

In my IE, it's work well

**Comment by nperriwp** [comment]
@AllenFang I have IE 11.0.9600.18762 on Windows 7, I see the default drop down of 10, 25, 30, 50 for "Fully Custom SizePerPage DropDown".
![2017-09-07 11_39_58-react bootstrap table - getting started - internet explorer](https://user-images.githubusercontent.com/9938647/30179476-6513d5e6-93c1-11e7-93e8-3cb9e26fc49f.png)

**Comment by AllenFang** [comment]
hmm.  there are any error log in the console log?

**Comment by nperriwp** [comment]
@AllenFang No, no errors in the console log.
